storage/decorator.py

```diff
@@ -3,78 +3,88 @@
 # you may not use this file except in compliance with the Elastic License 2.0.
 
 import zlib
-from functools import partial
 from typing import Any, Callable, Iterator
 
 from share import shared_logger
 
 from .storage import CommonStorageType
 
 
-class ByLines:
-    def __init__(
-        self, function: Callable[[CommonStorageType, int, int, Any, str, int], Iterator[tuple[bytes, int, int]]]
-    ):
-        self._function: Callable[
-            [CommonStorageType, int, int, Any, str, int], Iterator[tuple[bytes, int, int]]
-        ] = function
-
-        self._offset: int = 0
-        self._last_ending_offset: int = 0
-
-    def __call__(self, instance: CommonStorageType, *args: Any) -> Iterator[tuple[bytes, int, int]]:
+def by_lines(
+    func: Callable[[CommonStorageType, int, Any, str, int], Iterator[tuple[bytes, int, int]]]
```

Comment:
Would it be possible to define a type for this `Callable`? It looks error prone to use this in several places.

Reply:
oki for me